src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp

```diff
@@ -248,5 +256,182 @@ QVariantMap QgsLayoutAtlasToPdfAlgorithm::processAlgorithm( const QVariantMap &p
+
+    expression = parameterAsString( parameters, QStringLiteral( "FILTER_EXPRESSION" ), context );
+    atlas->setFilterExpression( expression, error );
+    atlas->setFilterFeatures( !expression.isEmpty() && error.isEmpty() );
+    if ( !expression.isEmpty() && !error.isEmpty() )
+    {
+      throw QgsProcessingException( QObject::tr( "Error setting atlas filter expression" ) );
+    }
+    error.clear();
+
+    expression = parameterAsString( parameters, QStringLiteral( "SORTBY_EXPRESSION" ), context );
+    if ( !expression.isEmpty() )
+    {
+      const bool sortByReverse = parameterAsBool( parameters, QStringLiteral( "SORTBY_REVERSE" ), context );
+      atlas->setSortFeatures( true );
+      atlas->setSortExpression( expression );
+      atlas->setSortAscending( !sortByReverse );
+    }
+    else
+    {
+      atlas->setSortFeatures( false );
+    }
+  }
+  else if ( !atlas->enabled() )
+  {
+    throw QgsProcessingException( QObject::tr( "Layout being export doesn't have an enabled atlas" ) );
+  }
+
+  const QgsLayoutExporter exporter( layout.get() );
+  QgsLayoutExporter::PdfExportSettings settings;
+
+  if ( parameters.value( QStringLiteral( "DPI" ) ).isValid() )
+  {
+    settings.dpi = parameterAsDouble( parameters, QStringLiteral( "DPI" ), context );
+  }
+  settings.forceVectorOutput = parameterAsBool( parameters, QStringLiteral( "FORCE_VECTOR" ), context );
+  settings.appendGeoreference = parameterAsBool( parameters, QStringLiteral( "GEOREFERENCE" ), context );
+  settings.exportMetadata = parameterAsBool( parameters, QStringLiteral( "INCLUDE_METADATA" ), context );
+  settings.simplifyGeometries = parameterAsBool( parameters, QStringLiteral( "SIMPLIFY" ), context );
+  settings.textRenderFormat = parameterAsEnum( parameters, QStringLiteral( "TEXT_FORMAT" ), context ) == 0 ? QgsRenderContext::TextFormatAlwaysOutlines : QgsRenderContext::TextFormatAlwaysText;
+
+  if ( parameterAsBool( parameters, QStringLiteral( "DISABLE_TILED" ), context ) )
+    settings.flags = settings.flags | QgsLayoutRenderContext::FlagDisableTiledRasterLayerRenders;
+  else
+    settings.flags = settings.flags & ~QgsLayoutRenderContext::FlagDisableTiledRasterLayerRenders;
+
+  settings.predefinedMapScales = QgsLayoutUtils::predefinedScales( layout.get() );
+
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "LAYERS" ), context );
+  if ( layers.size() > 0 )
+  {
+    const QList<QGraphicsItem *> items = layout->items();
+    for ( QGraphicsItem *graphicsItem : items )
+    {
+      QgsLayoutItem *item = dynamic_cast<QgsLayoutItem *>( graphicsItem );
+      QgsLayoutItemMap *map = dynamic_cast<QgsLayoutItemMap *>( item );
+      if ( map && !map->followVisibilityPreset() && !map->keepLayerSet() )
+      {
+        map->setKeepLayerSet( true );
+        map->setLayers( layers );
+      }
+    }
+  }
+  const QString filename = parameterAsString( parameters, QStringLiteral( "OUTPUT_FILENAME" ), context );
```

Comment:
Everything in this method from this line up is also shared code which could be moved to the base class.